sphinx_pyproject/__init__.py

```diff
@@ -273,10 +285,45 @@ def parse(
 		:param config:
 		:param set_defaults: Has no effect in this class.
 		"""
-
 		if "authors" in config:
 			config["author"] = config.pop("authors")
 		elif "maintainers" in config:
 			config["author"] = config.pop("maintainers")
 
 		return super().parse(config)
+
+
+class PoetryProjectParser(ProjectParser):
+
+	@staticmethod
+	def get_namespace(filename: PathPlus, config: Dict[str, TOML_TYPES]) -> Dict[str, TOML_TYPES]:
+		result = config.get("tool", {}).get("poetry")
+		if result is None:
+			raise BadConfigError(f"No 'tool.poetry' table found in {filename.as_posix()}")
+
+		return result
+
+	@staticmethod
+	def parse_author(config: Dict[str, TOML_TYPES]) -> str:
+		"""
+		Parse poetry's authors key.
+
+		:param config: The unparsed TOML config for the ``[tool.poetry]`` table.
+		"""
+
+		pep621_style_authors: List[dict[str, str]] = []
+
+		for author in config["author"]:
+			match = re.match(r"(?P<name>.*)<(?P<email>.*)>", author)
```

Comment:
I originally implemented this as a straight-copy of the base class's `parse_author`. The implementation used a simpler string splitting mechanism to **just** pull out the name, which is the required field for "author", iirc.

But given the 100% coverage requirement, I wanted to get reuse out of the preexisting test that handles lack of any authors at all, so it felt perhaps better to redirect back to the base class's implementation ultimately.

But I could just as easily add a test and have the simpler impl here.